### PR TITLE
[PROD-899] report name and version in X-Pusher-Library header

### PR DESCRIPTION
**Pull request by jameshfisher**

### CODE REVIEW

File: pusher.gemspec [hunk 1]
```diff
@@ -1,7 +1,13 @@
 # -*- encoding: utf-8 -*-
+
+# Copied from https://github.com/bundler/bundler/blob/524add155d6b90c679db21b03f0bd9f877f21ab0/bundler.gemspec#L3-L5
+# but made less cryptic
+$LOAD_PATH.unshift(File.expand_path('../lib/', __FILE__))
+require "pusher/version"
```
Comment (zimbatm) [inline pusher.gemspec]: Or load the file directly. Ruby 1.9+ resolves the file to the absolute file anyways.

```ruby
require File.expand_path('../lib/pusher/version', __FILE__)
```

Comment (zimbatm) [inline pusher.gemspec]: In 2.3+ it will be even nicer:
```
require File.join(__dir__, 'lib', 'pusher', 'version')
```

Comment (zimbatm) [inline pusher.gemspec]: There is also this version but require_relative is not reliably in an eval context (which rubygems tends to do)
```ruby
require_relative './lib/pusher/version'
```

Comment (jameshfisher) [inline pusher.gemspec]: I like this one: `require File.expand_path('../lib/pusher/version', __FILE__)`

I initially used `require_relative` but Ruby 1.9.3 didn't like it

Comment (zimbatm) [inline pusher.gemspec]: Yeah it's a shame `require_relative` isn't more reliable when all the information is clearly available in the context.

**Pull request merged**